### PR TITLE
Add API for Generating Invoice and Quotation Summary

### PR DESCRIPTION
**Pull request by Ando22**
# Context
This pull request introduces a new API endpoint that generates performance reports for both invoices and quotations relates with issues #226 . The reports provide valuable insights into the distribution of statuses over time, helping stakeholders make data-driven decisions.

# API Structure:
- Endpoint: `/api/{invoice / quote}/:type`
- Method: GET
- Parameters: type (optional): Defines the time period for the report. Possible values are "week," "month," or "year." If not specified, the default is "month."

## Response
```
{
    "success": true,
    "result": {
        "total": 11,
        "type": "week",
        "performance": [
            {
                "_id": "draft",
                "count": 6,
                "status": "draft",
                "percentage": 100
            },
            {
                "_id": "pending",
                "count": 1,
                "status": "pending",
                "percentage": 100
            },
            {
                "_id": "sent",
                "count": 4,
                "status": "sent",
                "percentage": 100
            }
        ]
    },
    "message": "Successfully found all invoices for the last week"
}
```

## How to Test:
To test the new API, follow these steps:
1. Clone this repository locally and switch to the feature branch.
2. Start the application and ensure it's running on your local server.
3. Use a tool like Postman or cURL to send a GET request to the API endpoint: Example: GET http://localhost:3000/api/{invoice / quote}/month
4. Replace "month" with "week" or "year" to generate reports for different time periods. Verify that the response contains the performance report in the specified time period, including the total count, type, and array of performance data for each status with count and percentage values.

### CODE REVIEW

File: controllers/erpControllers/invoiceController.js [hunk 1]
```diff
@@ -155,6 +156,86 @@ methods.update = async (req, res) => {
...
 };
 
+methods.summary = async (req, res) => {
+  try {
+    let defaultType = 'month';
+    
+    const { type } = req.query;
+
+    if (type) {
+      defaultType = type;
```
Comment (quan-nh2) [inline controllers/erpControllers/invoiceController.js:166]: I'm sorry for my unpolite comment, if not suitable for you, please just skip it.
What's happened if the ```defaultType``` is not week, month, or year? I think we need to catch and throw invalid params error here.

Comment (Ando22) [reply controllers/erpControllers/invoiceController.js:166]: No worries at all! You're right that handling cases where the defaultType parameter is not set to 'week', 'month', or 'year' is important. Thank you for pointing this out. I'll change the code.

Comment (quan-nh2) [reply controllers/erpControllers/invoiceController.js]: You are welcome

File: controllers/erpControllers/invoiceController.js [hunk 2]
```diff
@@ -155,6 +156,100 @@ methods.update = async (req, res) => {
...
+          }
+        },
+      },
+      {
+        $project: {
+          _id: 0,
+          status: '$_id',
+          count: 1,
+          percentage: {
+            $multiply: [{ $divide: ['$count', { $sum: '$count' }] }, 100],
```
Comment (salahlalami) [inline controllers/erpControllers/invoiceController.js:217]: @Ando22 please fix percentage , it doesn't calculate correctly , it always show 100

**Review by salahlalami — CHANGES_REQUESTED** [review]

File: controllers/erpControllers/invoiceController.js (hunk 2 above)
Comment (Ando22) [reply controllers/erpControllers/invoiceController.js]: @salahlalami Thank you for the noticing, fixed 🙏

**Review by salahlalami — APPROVED** [review]

**Review by polymahh — COMMENTED** [review]
i'll leave this to backend dev 
i tried to run the branch locally and hit the api endpoint but it was giving me an error instead of the json response

**Review by polymahh — APPROVED** [review]

**Pull request merged**